iceoryx_binding_c/test/moduletests/test_client.cpp

```diff
@@ -0,0 +1,492 @@
+            ResponseHeader(iox::cxx::UniqueId(), RpcBaseHeader::UNKNOWN_CLIENT_QUEUE_INDEX, 0U);
+        *static_cast<int64_t*>(chunk->getUserPayload()) = chunkValue;
+        iox::popo::ChunkQueuePusher<ClientChunkQueueData_t> pusher{&sutPort->m_chunkReceiverData};
+        pusher.push(*chunk);
+    }
+
+    void prepareClientInit(const ClientOptions& options = ClientOptions())
+    {
+        EXPECT_CALL(*runtimeMock,
+                    getMiddlewareClient(ServiceDescription{IdString_t(TruncateToCapacity, SERVICE),
+                                                           IdString_t(TruncateToCapacity, INSTANCE),
+                                                           IdString_t(TruncateToCapacity, EVENT)},
+                                        options,
+                                        _))
+            .WillOnce(Return(createClientPortData(options)));
+    }
+
+    bool isPayloadInDataSegment(const void* payload)
+    {
+        uint64_t startDataSegment = reinterpret_cast<uint64_t>(&dataMemory[0]);
+        uint64_t payloadPosition = reinterpret_cast<uint64_t>(payload);
+
+        return (startDataSegment <= payloadPosition && payloadPosition <= startDataSegment + DATA_MEMORY_SIZE);
+    }
+
+    static constexpr uint64_t MANAGEMENT_MEMORY_SIZE = 1024 * 1024;
+    char managementMemory[MANAGEMENT_MEMORY_SIZE];
+    iox::posix::Allocator mgmtAllocator{managementMemory, MANAGEMENT_MEMORY_SIZE};
+    static constexpr uint64_t DATA_MEMORY_SIZE = 1024 * 1024;
+    char dataMemory[DATA_MEMORY_SIZE];
+    iox::posix::Allocator dataAllocator{dataMemory, DATA_MEMORY_SIZE};
+    iox::mepoo::MemoryManager memoryManager;
+    iox::mepoo::MePooConfig memoryConfig;
+
+    iox::cxx::optional<ClientPortData> sutPort;
+    iox_client_storage_t sutStorage;
+
+    ServerChunkQueueData_t serverChunkQueueData{iox::popo::QueueFullPolicy::DISCARD_OLDEST_DATA,
+                                                iox::cxx::VariantQueueTypes::SoFi_MultiProducerSingleConsumer};
+    ChunkQueuePopper<ServerChunkQueueData_t> serverRequestQueue{&serverChunkQueueData};
+
+    static constexpr const char SERVICE[] = "allGlory";
+    static constexpr const char INSTANCE[] = "ToThe";
+    static constexpr const char EVENT[] = "HYPNOTOAD";
+};
+constexpr const char iox_client_test::RUNTIME_NAME[];
+constexpr const char iox_client_test::SERVICE[];
+constexpr const char iox_client_test::INSTANCE[];
+constexpr const char iox_client_test::EVENT[];
+
+TEST_F(iox_client_test, notInitializedOptionsAreUninitialized)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "347f3a6d-8659-4ac3-81be-720e8a444d5e");
+    iox_client_options_t uninitializedOptions;
+#if !defined(__clang__)
+    // ignore the warning since we would like to test the behavior of an uninitialized option
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+    EXPECT_FALSE(iox_client_options_is_initialized(&uninitializedOptions));
+#pragma GCC diagnostic pop
+#endif
```

Comment:
Do we really get a warning here?

Reply:
Yes, thats why I added this ignore here.